src/main/java/org/opentripplanner/model/TripIdAndServiceDate.java

```diff
@@ -0,0 +1,40 @@
+package org.opentripplanner.model;
+
+import java.util.Objects;
+import org.opentripplanner.model.calendar.ServiceDate;
+
+/**
+ * Class to use as key in HashMap containing feed id, trip id and service date
+ */
+public record TripIdAndServiceDate(FeedScopedId tripId, ServiceDate serviceDate) {
+  public FeedScopedId getTripId() {
+    return tripId;
+  }
+
+  public ServiceDate getServiceDate() {
+    return serviceDate;
+  }
```

Comment:
The `getTripId()` and `getServiceDate()` methods should be inlined. If you change the body to use the record accessor `tripId()`, then you can right click on the method and select _Refactor > Inline ..._.

Reply:
I agree - surprised the Intellij Record refactor didnt do this automatically. I have changed and inlined the field accessors now.